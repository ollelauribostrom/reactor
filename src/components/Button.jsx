import React from 'react';

type Props = {
  name: string,
  onClick: (e: SyntheticEvent<HTMLButtonElement>) => undefined
};

export default function Button({ name, onClick }: Props) {
  return (
    <button type="button" className="button" onClick={onClick}>
      {name}
    </button>
  );
}
