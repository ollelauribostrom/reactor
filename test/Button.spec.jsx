import React from 'react';
import { shallow } from 'enzyme';
import Button from '../src/components/Button';

describe('Tests for Button', () => {
  test('renders correctly', () => {
    const button = shallow(<Button name="test" onClick={() => jest.fn()} />);
    expect(button).toMatchSnapshot();
  });
  test('invokes onClick when clicked', () => {
    const onClick = jest.fn();
    const button = shallow(<Button name="test" onClick={onClick} />);
    button.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
