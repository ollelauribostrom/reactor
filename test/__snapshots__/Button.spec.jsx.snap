// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Button renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    name="test"
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "test",
      "className": "button",
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": "test",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
        "className": "button",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": "test",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
